trio/_tests/test_exports.py

```diff
@@ -264,111 +314,116 @@ def no_hidden(symbols):
-
-                if "conflicts with class variable access" in message:
-                    continue
-
-                errors[symbol] = error_type + ":" + message  # pragma: no cover
+            static_names -= ignore_names
 
         else:  # pragma: no cover
             assert False, "unknown tool"
 
+        missing = runtime_names - static_names
+        extra = static_names - runtime_names
+
+        # using .remove() instead of .delete() to get an error in case they start not
+        # being missing
+
+        if (
+            tool == "jedi"
+            and BaseException in class_.__mro__
+            and sys.version_info >= (3, 11)
+        ):
+            missing.remove("add_note")
+
+        if (
+            tool == "mypy"
+            and BaseException in class_.__mro__
+            and sys.version_info >= (3, 11)
+        ):
+            extra.remove("__notes__")
+
+        if tool == "mypy" and attrs.has(class_):
+            # e.g. __trio__core__run_CancelScope_AttrsAttributes__
+            before = len(extra)
+            extra = {e for e in extra if not e.endswith("AttrsAttributes__")}
+            assert len(extra) == before - 1
+
+        # TODO: this *should* be visible via `dir`!!
+        if tool == "mypy" and class_ == trio.Nursery:
+            extra.remove("cancel_scope")
+
+        # TODO: I'm not so sure about these, but should still be looked at.
+        EXTRAS = {
+            trio.DTLSChannel: {"peer_address", "endpoint"},
+            trio.DTLSEndpoint: {"socket", "incoming_packets_buffer"},
+            trio.Process: {"args", "pid", "stderr", "stdin", "stdio", "stdout"},
+            trio.SSLListener: {"transport_listener"},
+            trio.SSLStream: {"transport_stream"},
+            trio.SocketListener: {"socket"},
+            trio.SocketStream: {"socket"},
+            trio.testing.MemoryReceiveStream: {"close_hook", "receive_some_hook"},
+            trio.testing.MemorySendStream: {
+                "close_hook",
+                "send_all_hook",
+                "wait_send_all_might_not_block_hook",
+            },
+        }
+        if tool == "mypy" and class_ in EXTRAS:
+            before = len(extra)
+            extra -= EXTRAS[class_]
+            assert len(extra) == before - len(EXTRAS[class_])
+
+        # probably an issue with mypy....
+        if tool == "mypy" and class_ == trio.Path and sys.platform == "win32":
+            before = len(missing)
+            missing -= {"owner", "group", "is_mount"}
+            assert len(missing) == before - 3
+
+        # TODO: why is this? Is it a problem?
+        # see https://github.com/python-trio/trio/pull/2631#discussion_r1185615916
+        if class_ == trio.StapledStream:
+            extra.remove("receive_stream")
+            extra.remove("send_stream")
+
+        # I have not researched why these are missing, should maybe create an issue
+        # upstream with jedi
+        if tool == "jedi" and sys.version_info >= (3, 11):
```

Comment:
In case it isn't obvious, this changed from `sys.version_info >= (3, 12)` to `sys.version_info >= (3, 11)` cause of a backport, probably.